cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/ApplicationVerificationStage.java

```diff
@@ -183,6 +185,13 @@ private void verifyWorkflowSpecifications(ApplicationSpecification appSpec, Work
   private void verifyWorkflowNode(ApplicationSpecification appSpec, WorkflowSpecification workflowSpec,
                                   WorkflowNode node, Set<String> existingNodeNames) {
     WorkflowNodeType nodeType = node.getType();
+    if (node.getNodeId().equals(workflowSpec.getName())) {
```

Comment:
is this temporary until we have a better solution? If so we should add a todo with a jira.